### PR TITLE
SALTO-5343: convert statuses and transitions properties to a list

### PR DESCRIPTION
**Pull request by shayc331**
_convert statuses and transitions properties to a list_

---

_Additional context for reviewer_
* We use the same logic in other places in the adapter so I took it out as a more generic function

---
_Release Notes_: 
_None_

---
_User Notifications_: 
_None_

### CODE REVIEW

File: packages/jira-adapter/src/filters/workflowV2/workflow_filter.ts [hunk 1]
```diff
@@ -439,6 +443,10 @@ const filter: FilterCreator = ({ config, client, paginator, fetchQuery, elements
           const statusIdToUuid = getUuidMap(statusInstances)
           const resolvedWorkflowInstance = await resolveValues(workflowInstance, getLookUpName)
           convertTransitionParametersFields(resolvedWorkflowInstance.value.transitions, convertParametersFieldsToString)
+          convertPropertiesToMap([
+            ...resolvedWorkflowInstance.value.statuses ?? [],
+            ...resolvedWorkflowInstance.value.transitions ?? [],
+          ])
```
Comment (idoamit8) [inline packages/jira-adapter/src/filters/workflowV2/workflow_filter.ts:449]: I see that you are changing the struct of the filed in preDeploy and don't do anything in onDeploy.
Is it on purpose?

**Review by idoamit8 — COMMENTED** [review]
Great work! 💯 
added one question and a small nit :)

File: packages/jira-adapter/src/filters/workflowV2/workflow_filter.ts (hunk 1 above)
Comment (shayc331) [reply packages/jira-adapter/src/filters/workflowV2/workflow_filter.ts:449]: the preDeploy changes the element structure completely, I save the original element [here](https://github.com/salto-io/salto/blob/main/packages/jira-adapter/src/filters/workflowV2/workflow_filter.ts#L437) and change it back in the onDeploy [here](https://github.com/salto-io/salto/blob/main/packages/jira-adapter/src/filters/workflowV2/workflow_filter.ts#L476)

**Review by idoamit8 — APPROVED** [review]
LGTM 💯

**Pull request merged**